Added core support for message_to_radio

DanielEdriPurpl opened the pull request.

kantera800 commented on controller/src/beerocks/cli/beerocks_cli_bml.cpp:
```diff
@@ -1168,6 +1171,13 @@ int cli_bml::bml_channel_selection_caller(int numOfArgs)
 }
+int cli_bml::bml_message_to_radio_caller(int numOfArgs)
+{
+    if (numOfArgs == 2) {
```
this will never work , you should expect only 1 parameter

DanielEdriPurpl replied:
Big woopsie..

kantera800 commented on controller/src/beerocks/bml/bml.cpp:
```diff
@@ -584,6 +584,22 @@ int bml_trigger_topology_discovery(BML_CTX ctx, const char *al_mac)
+
+    bml_internal *pBML = (bml_internal *)ctx;
+
+    return (pBML->message_to_radio(radio_mac_addr));
```
there is missing implementation for` pBML->message_to_radio` , this commit cannot compile

kantera800 commented on common/beerocks/tlvf/yaml/beerocks/tlvf/beerocks_message_action.yaml:
```diff
@@ -109,7 +109,10 @@ eActionOp_CONTROL:
   ACTION_CONTROL_CHANNEL_SCAN_FINISHED_NOTIFICATION: 147
 
-  ACTION_CONTROL_ENUM_END: 148
+  ACTION_CONTROL_MESSAGE_TO_RADIO_REQUEST: 148
+  ACTION_CONTROL_MESSAGE_TO_RADIO_RESPONSE: 149
```
yamel and all related auto generated changes should be in separate "preparative" commit

kantera800 commented on controller/src/beerocks/bml/internal/bml_internal.cpp:
```diff
@@ -3480,6 +3480,30 @@ int bml_internal::channel_selection(const char *al_mac, const char *ruid)
     return BML_RET_OK;
 }
 
+int bml_internal::message_to_radio(const sMacAddr &radio_mac)
```
`message_to_radio` is used in previous commit.
either add it there  or add it as a separate "preparation" commit (without the yamel changes) and the next commit can actually use it

kantera800 commented on controller/src/beerocks/master/son_management.cpp:
```diff
@@ -1714,6 +1714,39 @@ void son_management::handle_bml_message(Socket *sd,
+        auto bml_request =
+            beerocks_header->addClass<beerocks_message::cACTION_BML_MESSAGE_TO_RADIO_REQUEST>();
+        if (bml_request == nullptr) {
+            LOG(ERROR) << "addClass CACTION_BML_MESSAGE_TO_RADIO_REQUEST failed";
```
```suggestion
            LOG(ERROR) << "addClass cACTION_BML_MESSAGE_TO_RADIO_REQUEST failed";
```

kantera800 commented on controller/src/beerocks/master/son_management.cpp:
```diff
@@ -1714,6 +1714,39 @@ void son_management::handle_bml_message(Socket *sd,
+            break;
+        }
+
+        std::string dst_mac = tlvf::mac_to_string(request->radio_mac());
```
you pulling the radio_mac from the wrong request 
```suggestion
        std::string dst_mac = tlvf::mac_to_string(bml_request->radio_mac());
```

kantera800 commented on controller/src/beerocks/master/son_management.cpp:
```diff
@@ -1714,6 +1714,39 @@ void son_management::handle_bml_message(Socket *sd,
+            break;
+        }
+
+        std::string dst_mac = tlvf::mac_to_string(request->radio_mac());
+        if (dst_mac == network_utils::WILD_MAC_STRING) {
+            auto agent_mac          = database.get_node_parent_ire(dst_mac);
+            const auto parent_radio = database.get_node_parent_radio(dst_mac);
+            son_actions::send_cmdu_to_agent(agent_mac, cmdu_tx, database, parent_radio);
+        }
```
dst_mac is already a radio mac you should just use it
```suggestion
        std::string radio = tlvf::mac_to_string(request->radio_mac());
        if (dst_mac == network_utils::WILD_MAC_STRING) {
            auto agent_mac          = database.get_node_parent_ire(radio );
            son_actions::send_cmdu_to_agent(agent_mac, cmdu_tx, database, radio);
        }
```

kantera800 commented on common/beerocks/tlvf/yaml/beerocks/tlvf/beerocks_message_bml.yaml:
```diff
@@ -529,3 +529,13 @@ cACTION_BML_CLIENT_GET_CLIENT_RESPONSE:
+  
+cACTION_BML_MESSAGE_TO_RADIO_REQUEST:
+  _type: class
+  radio_mac: sMacAddr
```
why did you add radio_mac if you are not using it ?

kantera800 commented on controller/src/beerocks/master/son_management.cpp:
```diff
@@ -1714,6 +1714,46 @@ void son_management::handle_bml_message(Socket *sd,
         }
         break;
     }
+    case beerocks_message::ACTION_BML_MESSAGE_TO_RADIO_REQUEST: {
```
should be separate commit

kantera800 commented on tests/test_flows.py:
```diff
@@ -454,6 +454,19 @@ def test_ap_config_bss_tear_down(self):
+            # Wait a bit for the message
+            time.sleep(1)
+            # Check controller
+            self.check_log(env.controller.mac, actions[0])
```
no need for `actions` code will be more readable  like this 
```suggestion
            self.check_log(env.controller.mac, ACTION_BML_MESSAGE_TO_RADIO_REQUEST)
```

kantera800 reviewed (commented):
1st commit:
- message:  better to use something like this
```
preparation commit.
message used for .....
```

-title:  is too long, should be  72 characters per line (you should see warning in vs code  gui) 
           should  include also location of the changes 
`yaml: bml:  add exercise  CMDUs & structs`

rest of commits:

- if you have changes all over the project  the order of the commits  should look like this:
```
yaml: bml: add CMDUs
bml: use CMDUs
yaml: controller: add CMDUs
controller: use CMDUs
yaml: agent: add CMDUs
agent: use CMDUs

```

DanielEdriPurpl committed:
yaml: bml: Adds CMDUs and a struct for 'message_to_radio'

Preparation commit
This message is used to send an 'acknowledgment' of some sort from a
controller towards its radios

DanielEdriPurpl committed:
bml: Use 'message_to_radio' CMDUs
Implementation of the CMDUs declared in the previous commit

DanielEdriPurpl committed:
yaml: controller: Add 'message_to_radio' CMDUs
This commit adds the control CMDUs for 'message_to_radio'

DanielEdriPurpl committed:
controller: use 'message_to_radio' CMDUs
This commit uses the CMDUs declared in the previous commit

DanielEdriPurpl committed:
agent: uses 'message_to_radio' CMDUs
This commit uses the CMDUs declared in the latest declarative commit,
but on the agent

DanielEdriPurpl committed:
test_flow.py: Adds 'message_to_radio' test
This adds a test to the test_flow.py script

DanielEdriPurpl committed:
yaml: bml: remove 60ghz TLVF references

This commit removes the 60ghz references in both .yaml files and the auto generated alike

DanielEdriPurpl committed:
controller: remove the 60ghz referneces

This commit removes the unsupported 60ghz references from the controller
